rep-2005.rst

```diff
@@ -0,0 +1,237 @@
+  * `ros2/rosidl_python <https://github.com/ros2/rosidl_python>`_
+  * `ros2/rosidl_runtime_py <https://github.com/ros2/rosidl_runtime_py>`_
+  * `ros2/rosidl_typesupport <https://github.com/ros2/rosidl_typesupport>`_
+
+* Interface definitions
+
+  * `ros-planning/navigation_msgs <https://github.com/ros-planning/navigation_msgs>`_
+  * `ros2/common_interfaces <https://github.com/ros2/common_interfaces>`_
+  * `ros2/example_interfaces <https://github.com/ros2/example_interfaces>`_
+  * `ros2/rcl_interfaces <https://github.com/ros2/rcl_interfaces>`_
+  * `ros2/test_interface_files <https://github.com/ros2/test_interface_files>`_
+  * `ros2/unique_identifier_msgs <https://github.com/ros2/unique_identifier_msgs>`_
+
+* RMW
+
+  * `ros2/rmw <https://github.com/ros2/rmw>`_
+  * `ros2/rmw_implementation <https://github.com/ros2/rmw_implementation>`_
+  * Connext (Connext itself is not open source)
+
+    * `ros2/rmw_connext <https://github.com/ros2/rmw_connext>`_
+    * `ros2/rosidl_typesupport_connext <https://github.com/ros2/rosidl_typesupport_connext>`_
+
+  * CycloneDDS
+
+    * `eclipse-cyclonedds/cyclonedds <https://github.com/eclipse-cyclonedds/cyclonedds>`_
+    * `ros2/rmw_cyclonedds <https://github.com/ros2/rmw_cyclonedds>`_
+
+  * FastRTPS
+
+    * `eProsima/Fast-CDR <https://github.com/eProsima/Fast-CDR>`_
+    * `eProsima/Fast-RTPS <https://github.com/eProsima/Fast-RTPS>`_
+    * `ros2/rmw_fastrtps <https://github.com/ros2/rmw_fastrtps>`_
+    * `ros2/rosidl_typesupport_fastrtps <https://github.com/ros2/rosidl_typesupport_fastrtps>`_
+
+  * OpenSplice
+
+    * `ros2/rmw_opensplice <https://github.com/ros2/rmw_opensplice>`_
+    * `ros2/rosidl_typesupport_opensplice <https://github.com/ros2/rosidl_typesupport_opensplice>`_
+
+* Client libraries
+
+  * `ros2/rcl <https://github.com/ros2/rcl>`_
+  * `ros2/rcl_logging <https://github.com/ros2/rcl_logging>`_
+  * `ros2/rclcpp <https://github.com/ros2/rclcpp>`_
+  * `ros2/rclpy <https://github.com/ros2/rclpy>`_
+
+* Orchestration
+
+  * `ros2/launch <https://github.com/ros2/launch>`_
+  * `ros2/launch_ros <https://github.com/ros2/launch_ros>`_
+
+* Features
+
+  * `ros/diagnostics <https://github.com/ros/diagnostics>`_
+  * `ros/xacro <https://github.com/ros/xacro>`_
+  * Sensor processing
+
+    * `ros-perception/image_common <https://github.com/ros-perception/image_common>`_
+    * `ros-perception/image_transport_plugins <https://github.com/ros-perception/image_transport_plugins>`_
+    * `ros-perception/perception_pcl <https://github.com/ros-perception/perception_pcl>`_
+    * `ros-perception/vision_opencv <https://github.com/ros-perception/vision_opencv>`_
+    * `ros-perception/laser_filters <https://github.com/ros-perception/laser_filters>`_
+    * `ros-perception/laser_geometry <https://github.com/ros-perception/laser_geometry>`_
+
+  * `ros-planning/navigation2 <https://github.com/ros-planning/navigation2>`_
+  * `ros-planning/moveit2 <https://github.com/ros-planning/moveit2>`_
```

Comment:
Autoware.Auto as a ROS2-based stack of algorithms for autonomous driving could also be listed: https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto.

Reply:
I am not sure adding AutowareAuto to this list is a good idea. While all these algorithms are very valuable for a specific domain I am not sure if the are relevant enough for a large enough part of the overall ROS community. Maybe others want to weight in on this...

Reply:
I agree with @dirk-thomas. I think that Autoware.Auto is:

- Still too niche
- Still too early in development

to be included in the ROS standard library.

Reply:
I'd argue agreement with @gbiggs but with the opposite conclusion. While adding it to the standard library *shipped* with ROS2, I'd agree, but I think that it should be included in this list so there's incentives for companies doing open-source work on Autoware to make it less niche or early so it can then it better used.

Reply:
The problem is that makes it increasingly difficult to decide where to draw the line, which will become increasingly important the more things this list gets used for. If we let Autoware in, then why not `Bob's Awesome Library for Pool-Playing Robots`?

Reply:
I think since Autoware is being developed by many companies, used by many people, and arguably the only form of open-source ROS-based framework for the class of robots that is autonomous driving being seriously developed, I think that meets my spec. If it were just 1 company primarily sponsoring it (which I believe there are a few) or just 1 primary user (which I think again there are a few), I'd agree. 

`Bob's Awesome Library` I don't think would make it if its just `Bob` developing it and primarily `Bob` using it.

This is an irritating suggestion, but maybe there should be a "ROS2 Standard Library Incubation" sublist, for projects that we want to count as TSC-representable projects, but aren't today mature enough to count as standalone projects.

Reply:
Hi guys,  lets then wait until this becomes wider adopted. We currently have 12 companies working on a large autonomous valet parking project due in April 2020 and I we can asses the suitability for inclusion into the standard library afterwards..